edc-extensions/control-plane-adapter/src/main/java/org/eclipse/tractusx/edc/cp/adapter/store/SqlQueueStore.java

```
@@ -0,0 +1,184 @@
+/*
+ * Copyright (c) 2022 ZF Friedrichshafen AG
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License, Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Contributors:
+ * ZF Friedrichshafen AG - Initial API and Implementation
+ *
+ */
+
+package org.eclipse.tractusx.edc.cp.adapter.store;
+
+import static org.eclipse.edc.sql.SqlQueryExecutor.executeQuery;
+import static org.eclipse.edc.sql.SqlQueryExecutor.executeQuerySingle;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.time.Clock;
+import java.time.Instant;
+import java.util.List;
+import java.util.Objects;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.eclipse.edc.spi.persistence.EdcPersistenceException;
+import org.eclipse.edc.sql.lease.SqlLeaseContextBuilder;
+import org.eclipse.edc.sql.store.AbstractSqlStore;
+import org.eclipse.edc.transaction.datasource.spi.DataSourceRegistry;
+import org.eclipse.edc.transaction.spi.TransactionContext;
+import org.eclipse.tractusx.edc.cp.adapter.dto.DataReferenceRetrievalDto;
+import org.eclipse.tractusx.edc.cp.adapter.store.model.QueueMessage;
+import org.eclipse.tractusx.edc.cp.adapter.store.schema.QueueStatements;
+
+public class SqlQueueStore extends AbstractSqlStore {
```

This class seems to be not tested, I would suggest at least a save/retrieve test against a sql server to verify mapping and queries.

---

It's not tested as I've cretaed only unit test for the classes with logic. SqlQueueStore is a utily class.

Integration tests are created within: https://jira.catena-x.net/browse/A1IDSC-393

Task is inprogress, and I believe there could be tests that make use of DB to test if reads/writes are correct

---

Business tests are a little far from the logic and hard to debug, I'm not in favor of delegating integration tests against DB to the E2E layer because this could become tedious

---

@ndr-brt 
But unit test are also not good place for testing integration with DB. 
Do You know some example in product-edc where this kind of DB test is cerated?

---

this is actually the first time we have a sql store in product edc, but in EDC they are well tested, e.g.:
https://github.com/eclipse-edc/Connector/blob/main/extensions/control-plane/store/sql/asset-index-sql/src/test/java/org/eclipse/edc/connector/store/sql/assetindex/PostgresAssetIndexTest.java

there the test suites are separated by meaning: unit tests, postgres integration tests, daps integration tests, ...
a good practice is to follow the "fail fast" pattern, furthermore business tests should test business, so e2e use cases, not integration.

---

Ok, I was not aware that there are test in EDC that uses real DB during test run. 
Looks to me that I can not use the "testFixtures" from core edc and the whole db setup. 
I understand that You expect me to set it all up for product-edc? 
I'm asking as I'm starting to have some doubts if this will be ready for next release.

---

I'm not really inclined to have features that aren't duly covered but, if we're sure that this works, good for me. I will just approve the PR, eager to see a subsequent one that covers the missing tests.

---

Thank You Andrea,
I really appreciate all Your comments, they are very usefull. I will try to add such test in future. Hope to find someone that knows how those EDC test works, as for now they are failing when I run them. Maybe there is some additional config that I've missed, but I would expect that running a method with @Test annotation should be enough...

---

to run them you should either add the `includeTags` property as it's done by the ci workflow:
https://github.com/eclipse-edc/Connector/blob/main/.github/workflows/verify.yaml#L212
or to comment the `@PostgresqlIntegrationTest` annotation.

I usually do the latter during development :)